lib/chef-init/process.rb

```
@@ -0,0 +1,123 @@
+#
+# Copyright:: Copyright (c) 2012-2014 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'sys/proctable'
+require 'chef-init/helpers'
+
+module ChefInit
+  class Process
+    include Sys
+    include ChefInit::Helpers
+
+    @@terminated_child_processes = {}
+    @@monitor_child_processes = []
+
+    def self.find_by_name(name)
+      Proctable.ps do |proc|
+        if proc.cmdline =~ process_name
+          return proc.pid
+        end
+      end
+      raise NotFound
+    end
+
+    def self.running?(pid)
+      if id.is_a? Integer
+        Proctable.ps(id).nil?
+      elsif id.is_a? String
+        Proctable.ps(get_pid(id)).nil?
+      else
+        false
```

Might be better to raise an error here.